os/include/assert.h

```diff
@@ -83,26 +84,30 @@
 #undef PANIC					/* Unconditional abort */
 
 #ifdef CONFIG_HAVE_FILENAME
-
+#define ASSERT_INFO(f, fmt, ...) \
+	{ if (!(f))	{ \
+			    snprintf(assert_info_str, CONFIG_STDIO_BUFFER_SIZE, fmt, ##__VA_ARGS__); \
+			    up_assert((const uint8_t *)__FILE__, (int)__LINE__, assert_info_str); \
```

Comment:
Do we need to deliver the string as an argument?
How about printing it at the assert function directly?
Because this buffer is dedicated to the assert.

Reply:
If we do as I suggested, we don't need to modify the up_assert format (arguments).

Reply:
Do you mean to directly print semaphore or other required information directly ?

Reply:
That's enough with snprintf. What I said is about up_assert. I want to keep the original up_assert.
```
#define ASSERT_INFO(f, fmt, ...) \
	{ if (!(f))	{ \
			    snprintf(assert_info_str, CONFIG_STDIO_BUFFER_SIZE, fmt, ##__VA_ARGS__); \
			    up_assert((const uint8_t *)__FILE__, (int)__LINE__); \

int up_assert(....)
{
  // printing assert_info_str by accessing global variable directly
}
```